examples/guide-supported-types-examples/src/non_null.rs

```diff
@@ -0,0 +1,13 @@
+use std::ptr;
+use std::ptr::NonNull;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub unsafe fn take_pointer_by_value(x: Option<NonNull<u8>>) {
+    Box::from_raw(x.unwrap().as_ptr());
+}
+
+#[wasm_bindgen]
+pub fn return_pointer() -> Option<NonNull<u8>> {
+    NonNull::from(Box::new(42).leak())
```

Comment:
I don't think this is valid: `Box::leak` can only be called as `Box::leak(x)`, not `x.leak()`, and this needs to either be wrapped in `Some` or not return an `Option`.

Also using `Box::leak` + `Box::from_raw` is a bit questionable, this should probably use `Box::into_raw` instead.

Reply:
Using [`Box::leak()` with `Box::from_raw()` is supported according to the documentation](https://doc.rust-lang.org/1.76.0/std/boxed/struct.Box.html#method.leak).

The reason why this is used is to get access to the infallible conversion to `NonNull`, compared to `NonNull::new()`. But in this case we return an `Option` anyway.